### PR TITLE
Error handling on the Rust side.

### PR DESCRIPTION
**Pull request by vext01**
Here's the error handling for the Rust side.

I follow up PR will simplify some of the error handling by re-structuring slightly the tracer context.

OK?

### CODE REVIEW

**Comment by vext01** [comment]
I've just realised that the error types not containing anything don't need to be tuples. Commit coming.

File: examples/simple_example.rs [hunk 1]
```diff
@@ -48,12 +48,19 @@ fn main() {
     let mut res = 0;
     let pid = unsafe { getpid() };
 
-    let mut tracer = Tracer::new().trace_filename("simple_example.ptt");
-    tracer.start_tracing();
+    let mut tracer = Tracer::new()
+                            .trace_filename("simple_example.ptt")
+                            .unwrap_or_else(|e| panic!(format!("{}", e)));
```
Comment (ltratt) [inline examples/simple_example.rs]: What's the difference between this line and just saying `unwrap()`?

File: src/lib.rs [hunk 2]
```diff
@@ -101,9 +104,9 @@ impl Tracer {
     /// # Arguments
     ///
     /// * `filename` - The filename in which to store trace packets.
-    pub fn trace_filename(mut self, filename: &str) -> Self {
-        self.trace_filename = CString::new(filename).expect("bad trace filename");
-        self
+    pub fn trace_filename(mut self, filename: &str) -> Result<Self, TraceMeError> {
```
Comment (ltratt) [inline src/lib.rs]: Is there a particular reason why `trace_filename` returns `Self`? Is it part of a "fluent" interface or ...?

**Comment by ltratt** [comment]
Although I think it's a bad idea, unfortunately `try!` has been superseded by `?`, so we should use that in new code. [I think the additional syntax was not a great idea, but such is life.]

File: examples/simple_example.rs (hunk 1 above)
Comment (vext01) [reply examples/simple_example.rs]: Ah, does `unwrap()` print the error message then? If so, we can just unwrap.

**Comment by vext01** [comment]
Ah. I'd forgotten to use `?`. It was introduced after v1 of the book was written. I can live with it, and may come to prefer it.

File: examples/simple_example.rs (hunk 1 above)
Comment (vext01) [reply examples/simple_example.rs]: It does. I pushed a change to simply unwrap.

**Comment by vext01** [comment]
Any further comments?

File: src/lib.rs (hunk 2 above)
Comment (vext01) [reply src/lib.rs]: I'm not sure what a fluent interface is, but I was following the rust paradigm where you construct and configure an object in a chain:

```
let mything = Thing().setting1("blah").setting2("flibble");
```

My next PR will simplify this particular piece of code, so there is not possibility of failure in the chain.

Comment (vext01) [reply src/lib.rs]: Looks like I just described a fluent interface :) Yes, a fluent interface, that's what I meant.

Comment (ltratt) [reply src/lib.rs]: OK. I just wanted to check you did it deliberately :)

**Comment by ltratt** [comment]
I think this is ready for squashing.

**Comment by vext01** [comment]
Like so.

**Comment by ltratt** [comment]
Just to be clear: this is squashed and ready for merging?

**Comment by vext01** [comment]
Correct. One commit is an unrelated fix, thus two commits.

**Pull request merged**